partials/seo.php

```diff
@@ -11,51 +11,54 @@
   echo '<meta name="fb:app_id" value="' . $fbAppId . '">';
 }
 
-// Getting values from current post [should this be in the is_single() section?]
-if (have_posts()) {
-  while (have_posts()) {
-    the_post();
-      $excerpt = get_the_excerpt();
-      if (has_post_thumbnail()) {
-        $thumb = wp_get_attachment_image_src( get_post_thumbnail_id($post->ID), 'opengraph' );
-      }
-  }
+?>
+  <meta property="og:title" content="<?php wp_title('|', true, 'right'); bloginfo('name'); ?>" />
+  <meta property="og:site_name" content="<?php bloginfo('name'); ?>" />
+<?php
+global $post;
+
+if (has_post_thumbnail($post)) {
+  $thumb = wp_get_attachment_image_src(get_post_thumbnail_id($post->ID), 'opengraph');
 }
 
-$ogImage = IGV_get_option('_igv_og_image');
+$ogImage = wp_get_attachment_image_src(IGV_get_option('_igv_og_image_id'), 'opengraph');
 
 if (!empty($thumb) && is_single()) {
   echo '<meta property="og:image" content="' . $thumb[0] . '" />';
 } else if (!empty($ogImage)) {
-  // I think here I need to get the correct size of this image from an ID?
-  echo '<meta property="og:image" content="' . $ogImage . '" />';
+  echo '<meta property="og:image" content="' . $ogImage[0] . '" />';
 } else {
   echo '<meta property="og:image" content="' . get_stylesheet_directory_uri() . '/img/dist/favicon.png" />';
 }
 
 if (is_home()) {
 ?>
   <meta property="og:url" content="<?php bloginfo('url'); ?>"/>
-  <meta property="og:title" content="<?php bloginfo('name'); ?>" />
-  <meta property="og:site_name" content="<?php bloginfo('name'); ?>" />
-  <meta property="og:type" content="website" />
   <meta property="og:description" content="<?php bloginfo('description'); ?>" />
   <meta name="twitter:card" value="<?php bloginfo('description'); ?>">
+  <meta property="og:type" content="website" />
 <?php
-} elseif (is_single()) {
+} else if (is_single()) {
+  global $post;
+  // trim post content by 600 chars
+  $excerpt = substr($post->post_content, 0, 600);
+  // strip html tags
+  $excerpt = strip_tags($excerpt);
```

Comment:
For some reason you can't use WP excerpt functions outside the loop thats why I did this, but perhaps there are lower level WP functions to generate this code rather than generic PHP ones